### DISCUSSION TITLE
Attention cell input shapes and Seq2Seq

### DISCUSSION THREAD

**Discussion by lambdaofgod** [dmlc/gluon-nlp#591] [Q&A] [answered]
How do Attention Cell's [call method](https://github.com/dmlc/gluon-nlp/blob/5db80c44ddff6b73eb7b746f5c438b13bc2b3d74/src/gluonnlp/model/attention_cell.py#L111) parameters map to parameters in Seq2Seq?

From what I understand in Seq2Seq decoder
* query corresponds to decoder states: this makes `(batch_size, query_length, query_dim)` equal to `(batch_size, output_sequence_length, num_hidden_decoder)`
* key corresponds to encoder outputs: `(batch_size, memory_length, key_dim)` is `(batch_size, input_sequence_length, num_hidden_encoder)`.
* value is just key, at least for dot and MLP attention cells.

Is my understanding right?

**Comment by szhengac** [comment, marked as answer]
Your understanding is correct.
